File: 2022/1A.py
#!/usr/bin/python
###
# 1A.
# [chrosta@toolbox:~/Github/advent-of-code@chrosta/advent-of-code/2022]$ cat ./data/1.text | ./1A.py 
###
import ast, sys
from itertools import groupby
data = ast.literal_eval(str([list(group) for k, group in groupby([l.strip() for l in sys.stdin.readlines()], lambda x: x == "") if not k]).replace("'", ""))
sums = [sum(d) for d in data]
sums.sort()
print(sums[-1:])
###
# 68787
###

File: 2024/1A.js
#! /usr/bin/node
//-------------------------------------------------------------------------------------------------------------
// 1A.js
//-------------------------------------------------------------------------------------------------------------
import ReadFile from './libs/read_file.js';

const rf = new ReadFile('./data/1.text');
rf.data((d) => {
    console.log("DATA:");
    console.log("---");
    Object.keys(d).forEach((k) => {
        console.log(d[k]);
    });
    console.log("---");

    let a = [];
    let b = [];
    let t = [];
    Object.keys(d).forEach((k) => {
        t = d[k].split(/\s+/);
        a.push(t[0]);
        b.push(t[1]);
    });
    a.sort();
    b.sort();
    console.log(a);
    console.log(b);
    console.log("---");
    
    let r = 0;
    let s = 0;
    Object.keys(a).forEach((i) => {
        r = Math.abs(a[i] - b[i]);
        s += r;
    });
    console.log(s);
});
